### ISSUE TITLE
Dead project?

### ISSUE THREAD

**Issue by jcroucher** [diiiary/capacitor-plugin-ios-swipe-back#6]
Looks like this project is dead. The developers site is also dead https://diiiary.com
Is there a fork we could point to as a new base?

**Comment by PowerOfM** [comment]
It probably is. The only line you really need is 
`bridge.getWebView()?.allowsBackForwardNavigationGestures = true;`

We ended up just making a "plugin" directly in the App folder that does this on load.

**Comment by ollyde** [comment]
@PowerOfM learning Capacitor here, where do I find 'bridge'. It doesn't say in the docs https://capacitorjs.com/docs/core-apis/ios

And it's not JS?

<img width="754" alt="Screenshot 2022-08-18 at 23 18 01" src="https://user-images.githubusercontent.com/6940726/185496850-3754c66e-4158-4430-90e3-ebd1fcc61cb3.png">

**Comment by ollyde** [comment]
For anyone else who got stuck.

<img width="1104" alt="Screenshot 2022-08-18 at 23 26 45" src="https://user-images.githubusercontent.com/6940726/185498179-77032a61-a0cf-47f3-a7cf-38148a8352a8.png">

**Comment by lmfmaier** [comment]
> For anyone else who got stuck.
> 
> <img alt="Screenshot 2022-08-18 at 23 26 45" width="1104" src="https://user-images.githubusercontent.com/6940726/185498179-77032a61-a0cf-47f3-a7cf-38148a8352a8.png">

This is not working for me. Do I need anything else?

**Comment by ollyde** [comment]
@lmfmaier it only works for swiping back in iOS.

**Comment by lmfmaier** [comment]
After some more research I came up with this code which works for me (forward and backward navigation gestures working in the ios web view)

In AppDelegate.swift

```
func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.
        (window?.rootViewController as? CAPBridgeViewController)?.loadViewIfNeeded()
        (window?.rootViewController as? CAPBridgeViewController)?.bridge?.webView?.allowsBackForwardNavigationGestures = true
        return true
    }
```

**Comment by ollyde** [comment]
@lmfmaier how does this handle links? It goes forwards? Do you have a video?

**Comment by lmfmaier** [comment]
> @lmfmaier how does this handle links? It goes forwards? Do you have a video?

What links do you mean? App/router links are handled by this. I just goes back and forward in the history. I use vue router in history mode.

I open external links in a separate browser window rand not in the web view.

**Comment by ollyde** [comment]
@lmfmaier we're talking about capacitor here and default web-views as we are using Svelte and not including any framework supplied by Ionic.

How does this handle default links when it moves forward? Does it slide to the next screen?

**Comment by lmfmaier** [comment]
> @lmfmaier we're talking about capacitor here and default web-views as we are using Svelte and not including any framework supplied by Ionic.
> 
> How does this handle default links when it moves forward? Does it slide to the next screen?

A "regular" link that opens a new document in the same web view's history is handled by this, just like in Safari.

**Comment by ollyde** [comment]
@lmfmaier ah ok so i doesn't fix the issue. We'll stick to Flutter 😂

**Comment by sergiusvet** [comment]
> After some more research I came up with this code which works for me (forward and backward navigation gestures working in the ios web view)
> 
> In AppDelegate.swift
> 
> ```
> func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
>         // Override point for customization after application launch.
>         (window?.rootViewController as? CAPBridgeViewController)?.loadViewIfNeeded()
>         (window?.rootViewController as? CAPBridgeViewController)?.bridge?.webView?.allowsBackForwardNavigationGestures = true
>         return true
>     }
> ```

This actually works. Only it has a strange issue/behavior. We use Vue + Vuetify, and the router-view is surrounded by a transition. And the transition is buggy on iOS. On Google Pixel it works fine btw.

**Comment by ollyde** [comment]
@sergiusvet it doesn't work going forwards though which makes the app seem very janky.

**Comment by sergiusvet** [comment]
@ollydixon Nah, I tested my app in Safari, and it has nothing to do with this approach, the app is "janky" there too, and I suppose it's because of the combination of how the vuetify + vue router transition works and how safari handles back/forward navigation animation.

**Comment by ollyde** [comment]
@sergiusvet I mentioned before; not using and wrappers for navigation animations; would be nice to get this native; I guess that would require multiple webviews and a native overlying :-).

**Comment by khromov** [comment]
Thanks @ollyde and @sergiusvet for your excellent examples!

I am trying sergiusvet last version of the code and it works great, except that the user can go back from the first page the landed on to a completely white page. Then a few seconds later you go back to the page you were previously on, has any of you run into this issue and know how you can prevent navigating to this empty blank page?

**Comment by ollyde** [comment]
@khromov it won't go back if it's the last page :-)
